pkg/spec/spec_test.go

```diff
@@ -26,14 +28,28 @@ func makeTestCreateConfig() *CreateConfig {
 	return cc
 }
 
-// TestPIDsLimit verifies the given pid-limit is correctly defined in the spec
-func TestPIDsLimit(t *testing.T) {
+func doCommonSkipChecks(t *testing.T) {
 	// The default configuration of podman enables seccomp, which is not available on non-Linux systems.
 	// Thus, any tests that use the default seccomp setting would fail.
 	// Skip the tests on non-Linux platforms rather than explicitly disable seccomp in the test and possibly affect the test result.
 	if runtime.GOOS != "linux" {
 		t.Skip("seccomp, which is enabled by default, is only supported on Linux")
 	}
+
+	isUnified, err := cgroups.IsCgroup2UnifiedMode()
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	if rootless.IsRootless() && !isUnified {
```

Comment:
Should do the rootless check first, since you would not need to do the Unified check if not rootless.

I hate the isUnified variable, could you use isCgroupV2, to make it clearer what you are checking.

Reply:
Fixed both items.